services/spar/test/Test/Spar/Sem/IdPSpec.hs

```diff
@@ -9,64 +10,237 @@ import Imports
+  PropConstraints r f
+
+instance
+  (Member E.IdP r, forall z. Show z => Show (f z), forall z. Eq z => Eq (f z)) =>
+  PropConstraints r f
+
+prop_storeStore ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_storeStore =
+  prepropLaw @'[E.IdP] $ do
+    s <- arbitrary
+    s' <- arbitrary
+    pure
+      ( do
+          E.storeConfig $ s & SAML.idpId .~ s' ^. SAML.idpId
+          E.storeConfig s',
+        do
+          E.storeConfig s'
+      )
+
+prop_storeStoreInterleave ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_storeStoreInterleave =
+  prepropLaw @'[E.IdP] $ do
+    s <- arbitrary
+    s' <- arbitrary
+    !_ <-
+      if s ^. SAML.idpId == s' ^. SAML.idpId
+        then discard
+        else pure ()
+    pure
+      ( do
+          E.storeConfig s
+          E.storeConfig s',
+        do
+          E.storeConfig s'
+          E.storeConfig s
+      )
+
+prop_storeGet ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_storeGet =
+  prepropLaw @'[E.IdP] $
+    do
+      s <- arbitrary
+      pure
+        ( do
+            E.storeConfig s
+            E.getConfig $ s ^. idpId,
+          do
+            E.storeConfig s
+            pure (Just s)
+        )
+
+prop_deleteGet ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_deleteGet =
+  prepropLaw @'[E.IdP] $ do
+    s <- arbitrary
+    pure
+      ( do
+          E.deleteConfig s
+          E.getConfig $ s ^. SAML.idpId,
+        do
+          E.deleteConfig s
+          pure Nothing
+      )
+
+prop_deleteDelete ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_deleteDelete =
+  prepropLaw @'[E.IdP] $ do
+    s <- arbitrary
+    pure
+      ( do
+          E.deleteConfig s
+          E.deleteConfig s,
+        do
+          E.deleteConfig s
+      )
+
+prop_storeGetByIssuer ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_storeGetByIssuer =
+  prepropLaw @'[E.IdP] $
+    do
+      s <- arbitrary
+      pure
+        ( do
+            E.storeConfig s
+            E.getIdByIssuerWithoutTeam $ s ^. idpMetadata . edIssuer,
+          do
+            E.storeConfig s
+            pure $ E.GetIdPFound $ s ^. idpId
+        )
+
+prop_setClear ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_setClear =
+  prepropLaw @'[E.IdP] $
+    do
+      replaced <- arbitrary
+      replacing <- arbitrary
+      pure
+        ( do
+            E.setReplacedBy replaced replacing
+            E.clearReplacedBy replaced,
+          do
+            E.clearReplacedBy replaced
+        )
+
+prop_getGet ::
+  PropConstraints r f =>
+  (forall a. Sem r a -> IO (f a)) ->
+  Property
+prop_getGet =
+  prepropLaw @'[E.IdP] $
+    do
+      idpid <- arbitrary
+      pure
+        ( do
+            liftA2 (,) (E.getConfig idpid) (E.getConfig idpid),
```

Comment:
Are you sure this is ever going to pull anything?  I think I've asked this question once above, so I'll stop now.  :)

Reply:
see above.